www/front_src/src/Navigation/Sidebar/Menu/CollapsItem.tsx

```diff
@@ -0,0 +1,292 @@
+  },
+  root: {
+    '& .MuiListItemIcon-root': {
+      minWidth: theme.spacing(2.25),
+      padding: theme.spacing(0, 0.5, 0, 1),
+    },
+    '& .MuiTypography-root': {
+      color: theme.palette.text.primary,
+      fontSize: theme.typography.caption,
+    },
+    border: `solid ${theme.palette.divider} 0.1px`,
+  },
+  scroll: {
+    '&::-webkit-scrollbar': {
+      width: theme.spacing(1.2),
+    },
+    '&::-webkit-scrollbar-thumb': {
+      backgroundColor: theme.palette.action.disabled,
+    },
+    '&::-webkit-scrollbar-track': {
+      border: `solid ${theme.palette.action.hover} 0.5px`,
+    },
+    maxHeight: theme.spacing(44),
+    overflowX: 'hidden',
+  },
+  subHeader: {
+    color: theme.palette.text.secondary,
+    fontSize: theme.typography.body2.fontSize,
+    fontWeight: 'bold',
+    lineHeight: theme.spacing(3),
+    textAlign: 'center',
+  },
+  toggled: {
+    backgroundColor: theme.palette.background.default,
+    left: ({ currentWidth }: StyleProps): string => theme.spacing(currentWidth),
+    minWidth: collapsWidth,
+    position: 'fixed',
+    top: ({ currentTop }: StyleProps): number | undefined => currentTop,
+    zIndex: theme.zIndex.mobileStepper,
+  },
+}));
+
+const CollapsItem = ({
+  data,
+  isCollapsed,
+  isSubHeader,
+  currentTop,
+  currentWidth,
+  onClick,
+  level,
+}: CollapsProps): JSX.Element => {
+  const classes = useStyles({ currentTop, currentWidth });
+  const [hoveredIndex, setHoveredIndex] = useState<number | null>(null);
+  const [topItem, setTopItem] = useState<number>();
+  const [navigationItemSelected, setNavigationItemSelected] = useAtom(
+    navigationItemSelectedAtom,
+  );
+  const levelName = `level_${level}_Navigated`;
+  const widthItem = currentWidth + collapsWidth / 8 + 0.15;
+
+  const handleHover = (
+    e: React.MouseEvent<HTMLElement>,
+    index: number,
+    item: Page,
+  ): void => {
+    const rect = e.currentTarget.getBoundingClientRect();
+    const top = rect.bottom - rect.height;
+    setTopItem(top);
+    setHoveredIndex(index);
+    const levelLabel = `level_${level}`;
+
+    setNavigationItemSelected({
+      ...navigationItemSelected,
+      [levelLabel]: { index, label: item.label, url: item?.url },
+    });
+  };
+
+  const handleLeave = (): void => {
+    setHoveredIndex(null);
+    if (navigationItemSelected) {
+      Object.keys(navigationItemSelected).forEach(() => {
+        delete navigationItemSelected[`level_${level}`];
+      });
+    }
+  };
+
+  const isHover = (
+    object: Record<string, propsNavigationItemSelected> | null,
+    levelTitle: string,
+    index: number,
+    item: Page,
+  ): boolean => {
+    if (object && object[levelTitle]) {
+      return (
+        object[levelTitle].index === index &&
+        object[levelTitle].label === item.label &&
+        object[levelTitle].url === item?.url
+      );
+    }
+
+    return false;
+  };
+
+  const getNestedIndex = (index, ind: number, content: Array<Page>): number => {
```

Comment:
It's a bit confusing to have to parameters with almost the same names: index and ind. Can we find better names?